crates/p2p/src/client/peer_agnostic.rs

```diff
@@ -500,6 +546,130 @@ impl Client {
         }
         }
     }
+
+    /// ### Important
+    ///
+    /// Events are grouped by block and by transaction. The order of flattened events in a block is guaranteed
+    /// to be correct because the event commitment is part of block hash. However the number of events per transaction
+    /// for __pre 0.13.2__ Starknet blocks is __TRUSTED__ because neither signature nor block hash contain this information.
```

Comment:
I'm pretty sure this is correct... the [pre-0.13.2 signature](https://community.starknet.io/t/introducing-p2p-authentication-and-mismatch-resolution-in-v0-12-2/97993) does not cover it, neither does the block hash.

Reply:
For example we just trust that for any pre-0.13.2 block:
```
(txn0, [eventA, eventB]),(txn1, [eventC])
```
is correct and not for example:
```
(txn0, [eventA]),(txn1, [eventB, eventC])
```

Reply:
Ofc we won't have this ambiguity if the >=0.13.2 block hash could be computed and verified for pre 0.13.2 blocks.